src/webhook/mutation/pod_mutator.go

```diff
@@ -751,14 +752,31 @@ func addMetadataIfMissing(c *corev1.Container, deploymentMetadata *deploymentmet
 			return
 		}
 	}
-
 	c.Env = append(c.Env,
 		corev1.EnvVar{
 			Name:  dynatraceMetadataEnvVarName,
 			Value: deploymentMetadata.AsString(),
 		})
 }
 
+func setInitialConnectRetryIfMissing(c *corev1.Container, dynaKube *dynatracev1beta1.DynaKube) {
+	if dynaKube.FeatureAgentInitialConnectRetry() < 0 {
+		return
+	}
+
+	for _, v := range c.Env {
```

Comment:
Don't override if manually set by user? Please extract it to a method with descriptive name